tests/baselines/reference/circularReferenceInReturnType2.types

```diff
@@ -104,8 +104,8 @@ type Something = { foo: number };
 
 // inference fails here, but ideally should not
 const A = object<Something>()({
->A : ObjectType<Something>
->  : ^^^^^^^^^^^^^^^^^^^^^
+>A : any

```

Comment:
The previous type was better here but i guess maybe that relied on broken assumptions from the PR that got reverted

Reply:
Right, when we report circularity errors we also assign the error type, but the reverted PR sometimes didn't do that.